lib/pbench/server/api/resources/__init__.py

```diff
@@ -195,6 +195,19 @@ def __str__(self):
         return f"Unrecognized list value{'s' if len(self.bad) > 1 else ''} {self.bad!r} given for parameter {self.parameter.name}; expected {expected}"
 
 
+class APIAbort(Exception):
+    """
+    Used to report an Invalid API Request
+    """
+
+    def __init__(self, http_status: int, message: str):
+        self.http_status = http_status
+        self.message = message
```

Comment:
As a suggestion, we could supply a default value for the message string based on the status value:
```
    def __init__(self, http_status: int, message: str = None):
        self.http_status = http_status
        self.message = message if message else HTTPStatus(http_status).phrase
```
In this way, we can declare an `APIAbort(500)` and it will automatically have a message of `'Internal Server Error'` without us having to spell it out.  (That is, when we don't have anything interesting to say, we don't have to say anything.)